core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4548,29 +4548,42 @@ public Object getTarget() {
         try {
             checkPermission(READ);
         } catch (AccessDeniedException e) {
-            String rest = Stapler.getCurrentRequest().getRestOfPath();
-            for (String name : ALWAYS_READABLE_PATHS) {
-                if (rest.startsWith(name)) {
-                    return this;
-                }
-            }
-            for (String name : getUnprotectedRootActions()) {
-                if (rest.startsWith("/" + name + "/") || rest.equals("/" + name)) {
-                    return this;
-                }
-            }
-
-            // TODO SlaveComputer.doSlaveAgentJnlp; there should be an annotation to request unprotected access
-            if (rest.matches("/computer/[^/]+/slave-agent[.]jnlp")
-                && "true".equals(Stapler.getCurrentRequest().getParameter("encrypt"))) {
+            if (isPathUnprotected(Stapler.getCurrentRequest().getRestOfPath())) {
                 return this;
             }
 
-
             throw e;
         }
         return this;
     }
+    
+    /**
+     * Test a path to see if it does not require authentication
```

Comment:
I'm having a hard time coming up with a name that is simple and not potentially misleading...

The basic idea it needs to convey is that the paths this function checks for are outside of the normal security settings for Jenkins, and are generally accessible regardless of authentication status, but we are not checking anything that can be controlled by the normal security settings.

`isPathAlwaysReadable` seems to be pretty close to the truth for the paths I know of, but I'm not sure how true this may be for `UnprotectedRootAction`s that implement authentication requirements themselves, and seems too close to `ALWAYS_READABLE_PATHS` for something that checks more than just that collection.

Even better: `isPathAccessControlExempt`, since that seems to capture the point of `UnprotectedRootActions` and `ALWAYS_READABLE_PATHS`, with low chance of confusion; a longer and more clear name: `isPathExemptFromContainerAccessControl` - infers that there may still be some access control on it, but that access control is independent from the containers' access control.

Not looking for someone else to do the work, instead looking for some guidance before I commit changes to this pull request.